spec/timezone.html

```diff
@@ -5,6 +5,98 @@
+    </emu-note>
+
+    <emu-clause id="sec-hostisvalidtimezonename" aoid="HostIsValidTimeZoneName">
+      <h1>HostIsValidTimeZoneName ( _timeZone_ )</h1>
+
+      <p>
+        HostIsValidTimeZoneName is a host-defined abstract operation that takes argument _timeZone_ (a String value).
+        It verifies that _timeZone_ represents a valid built-in time zone.
+      </p>
+      <p>
+        The implementation of HostIsValidTimeZoneName must conform to the following requirements:
+      </p>
+      <ul>
+        <li>It must always complete normally (i.e., not return an abrupt completion.)</li>
+        <li>It must always return either *true* or *false*.</li>
+        <li>Once it has returned *true* for a particular value of _timeZone_, for the lifetime of the surrounding agent, it may no longer return *false* for that value of _timeZone_, or for the value of HostCanonicalizeTimeZoneName(_timeZone_).
+      </ul>
+      <p>
+        The implementation of HostIsValidTimeZoneName should return *true* if _timeZone_, converted to upper case, is equal to one of the built-in time zone names, converted to upper case.
+        It should return *false* otherwise.
+      </p>
+      <p>The minimum implementation of HostIsValidTimeZoneName, supporting only the *"UTC"* time zone, performs the following steps when called:
+      </p>
+
+      <emu-alg>
+        1. Assert: Type(_timeZone_) is String.
+        1. Let _tzText_ be ! StringToCodePoints(_timeZone_).
+        1. Let _tzUpperText_ be the result of toUppercase(_tzText_), according to the Unicode Default Case Conversion algorithm.
+        1. Let _tzUpper_ be ! CodePointsToString(_tzUpperText_).
+        1. If _tzUpper_ and *"UTC"* are the same sequence of code points, return *true*.
+        1. Return *false*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-hostcanonicalizetimezonename" aoid="HostCanonicalizeTimeZoneName">
+      <h1>HostCanonicalizeTimeZoneName ( _timeZone_ )</h1>
+
+      <p>
+        HostCanonicalizeTimeZoneName is a host-defined abstract operation that takes argument _timeZone_ (a String value).
+        It returns the canonical and case-regularized form of the _timeZone_ argument.
+      </p>
+      <p>
+        HostCanonicalizeTimeZoneName may only be called with a _timeZone_ parameter that is a valid time zone name according to HostIsValidTimeZoneName.
+      </p>
+      <p>
+        The implementation of HostCanonicalizeTimeZoneName must conform to the following requirements:
+      </p>
+      <ul>
+        <li>It must always complete normally (i.e., not return an abrupt completion.)</li>
+        <li>It must always return a String value.</li>
+      </ul>
+      <p>The minimum implementation of HostCanonicalizeTimeZoneName, supporting only the *"UTC"* time zone, performs the following steps when called:
+      </p>
+
+      <emu-alg>
+        1. Assert: Type(_timeZone_) is String.
+        1. Assert: ! HostIsValidTimeZoneName(_timeZone_) is *true*.
```

Comment:
Generally the `?` and `!` are needed at the callsite regardless. However, for predicates, we've occasionally left them off since they don't ever return abruptly. For now, probably best to keep them in.